pkg/clustermesh/remote_cluster.go

```diff
@@ -275,6 +277,53 @@ func (rc *remoteCluster) restartRemoteConnection(allocator RemoteIdentityWatcher
 	)
 }
 
+func (rc *remoteCluster) getClusterConfig(ctx context.Context, backend kvstore.BackendOperations, forceRequired bool) (*cmtypes.CiliumClusterConfig, error) {
+	var (
+		err                       error
+		config                    *cmtypes.CiliumClusterConfig
+		requireConfig             = forceRequired
+		clusterConfigPollInterval = 30 * time.Second
+		clusterConfigPollTimeout  = 3 * time.Minute
+	)
+
+	if !requireConfig {
+		// Let's check whether the kvstore states that the cluster configuration should be always present.
+		requireConfig, err = IsClusterConfigRequired(backend)
+		if err != nil {
+			return nil, fmt.Errorf("failed to detect whether the cluster configuration is required: %w", err)
+		}
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, clusterConfigPollTimeout)
+	defer cancel()
+
+	// We retry here rather than returning an simply returning an error and relying on the
+	// controller backoff period to avoid recreating every time a new connection to the remote
+	// kvstore, which would introduce an unnecessary overhead. Still, we do return in case of
+	// consecutive failures, to ensure that we do not retry forever is something strange happened.
+	err = wait.PollImmediateInfiniteWithContext(ctx, clusterConfigPollInterval, func(ctx context.Context) (done bool, err error) {
```

Comment:
We don't need the name for return value?

Reply:
Yeah, I agree that it is not strictly required, but IMHO it helps a bit clarifying the meaning of the boolean parameter.